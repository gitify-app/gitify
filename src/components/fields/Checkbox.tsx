import type { FC, ReactNode } from 'react';
import { Tooltip } from './Tooltip';

export interface ICheckbox {
  name: string;
  label: string;
  helpText?: ReactNode | string;
  tooltip?: ReactNode | string;
  checked: boolean;
  disabled?: boolean;
  onChange: (evt: React.ChangeEvent<HTMLInputElement>) => void;
}

export const Checkbox: FC<ICheckbox> = (props: ICheckbox) => {
  return (
    <div className="mb-3 mt-1 text-sm">
      <div className="flex items-start">
        <div className="flex h-5 items-center">
          <input
            type="checkbox"
            id={props.name}
            className="size-4 rounded border-gray-300 text-indigo-600 focus:ring-indigo-500"
            checked={props.checked}
            onChange={props.onChange}
            disabled={props.disabled}
          />
        </div>

        <div className="ml-3">
          <label
            htmlFor={props.name}
            className="font-medium text-gray-700 dark:text-gray-200"
            style={
              props.disabled ? { textDecoration: 'line-through' } : undefined
            }
          >
            {props.label}
            {props.tooltip && (
              <Tooltip name={`tooltip-${props.name}`} tooltip={props.tooltip} />
            )}
          </label>
        </div>
      </div>

      {props.helpText && (
        <div className="mt-1 text-xs italic text-gray-500 dark:text-gray-300">
          {props.helpText}
        </div>
      )}
    </div>
  );
};
