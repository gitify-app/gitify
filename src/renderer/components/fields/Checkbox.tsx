import type { FC, ReactNode } from 'react';

import { CounterLabel, Stack } from '@primer/react';

import { cn } from '../../utils/cn';
import { Tooltip } from './Tooltip';

export interface ICheckbox {
  name: string;
  label: string;
  counter?: number;
  tooltip?: ReactNode | string;
  checked: boolean;
  disabled?: boolean;
  visible?: boolean;
  onChange: (evt: React.ChangeEvent<HTMLInputElement>) => void;
}

export const Checkbox: FC<ICheckbox> = ({
  visible = true,
  ...props
}: ICheckbox) => {
  const counter = props.counter > 0 ? props.counter : undefined;

  return (
    visible && (
      <Stack
        direction="horizontal"
        gap="condensed"
        align="center"
        className="text-sm"
      >
        <input
          type="checkbox"
          id={props.name}
          className="size-4 rounded-sm cursor-pointer"
          checked={props.checked}
          onChange={props.onChange}
          disabled={props.disabled}
          data-testid={`checkbox-${props.name}`}
        />

        <label
          htmlFor={props.name}
          className={cn(
            'font-medium text-gitify-font cursor-pointer',
            props.disabled && 'line-through',
          )}
        >
          {props.label}
        </label>

        {props.tooltip && (
          <Tooltip name={`tooltip-${props.name}`} tooltip={props.tooltip} />
        )}

        {counter && (
          <CounterLabel scheme={props.checked ? 'primary' : 'secondary'}>
            {counter}
          </CounterLabel>
        )}
      </Stack>
    )
  );
};
