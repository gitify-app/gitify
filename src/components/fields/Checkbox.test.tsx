import * as TestRenderer from 'react-test-renderer';

import { Checkbox, type ICheckbox } from './Checkbox';

describe('components/fields/Checkbox.tsx', () => {
  const props: ICheckbox = {
    name: 'appearance',
    label: 'Appearance',
    helpText: 'This is some helper text',
    checked: true,
    onChange: jest.fn(),
  };

  it('should render', () => {
    const tree = TestRenderer.create(<Checkbox {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
