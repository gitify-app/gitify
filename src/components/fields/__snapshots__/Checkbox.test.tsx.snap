// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/fields/Checkbox.tsx should render 1`] = `
<div
  className="mt-1 mb-3 text-sm"
>
  <div
    className="flex items-start"
  >
    <div
      className="flex items-center h-5"
    >
      <input
        checked={true}
        className="focus:ring-indigo-500 h-4 w-4 text-indigo-600 border-gray-300 rounded"
        id="appearance"
        onChange={[MockFunction]}
        type="checkbox"
      />
    </div>
    <div
      className="ml-3 "
    >
      <label
        className="font-medium text-gray-700 dark:text-gray-200"
        htmlFor="appearance"
      >
        Appearance
      </label>
    </div>
  </div>
  <div
    className="text-xs mt-1 italic text-gray-500 dark:text-gray-300"
  >
    This is some helper text
  </div>
</div>
`;
